src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -482,7 +483,6 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
             if (this.props.teachSession.dialogMode !== CLM.DialogMode.EndSession) {
 
                 if (save) {
-
                     // If editing an existing train dialog, extract its dialogId
                     const sourceTrainDialogId = this.sourceTrainDialogId()
 
```

Comment:
Won't let me add comment on the lines below, but this is the other location to add the `editTrainDialogThunkAsync` call.
Would call it every time instead of only on no merge.